arwen/elrondapi/elrondei.go

```diff
@@ -1732,10 +1647,105 @@ func getOriginalTxHash(context unsafe.Pointer, dataOffset int32) {
 	_ = arwen.WithFault(err, context, runtime.ElrondAPIErrorShouldFailExecution())
 }
 
-func areInSameShard(host arwen.VMHost, leftAddress []byte, rightAddress []byte) bool {
-	blockchain := host.Blockchain()
-	leftShard := blockchain.GetShardOfAddress(leftAddress)
-	rightShard := blockchain.GetShardOfAddress(rightAddress)
+func prepareIndirectContractCallInput(
```

Comment:
this PR has some changes from the previous one which is not yet merged. let's do them step by step.

Reply:
Indeed, this depends on https://github.com/ElrondNetwork/arwen-wasm-vm/pull/213 must be merged first.